src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java

```diff
@@ -73,15 +76,26 @@ public int getGeneratedOutput(Location l, Config data) {
         final boolean canGenerate = stored < getCapacity();
         final int rate = canGenerate ? getGeneratingAmount(inv.getBlock(), l.getWorld()) : 0;
 
+        if (rate == this.type.getDayGenerationRate()) {
+            generationType = "&aOverworld &e(Day)";
+        } if (rate == this.type.getNightGenerationRate()){
+            generationType = "&aOverworld &8(Night)";
+        } if (l.getWorld().getEnvironment() == World.Environment.NETHER){
+            generationType = "&cNether &e(Day)";
+        } if (l.getWorld().getEnvironment() == World.Environment.THE_END){
+            generationType = "&5End &8(Night)";
```

Comment:
These (end, nether) should be above the generation rate

Reply:
Why? Then it will find end, then find its day, and then display overworld.
its ifs, not else ifs

Reply:
I could change them to else ifs if you would like

Reply:
Why are they not else ifs to begin with NotLikeThis

Formatted that way